array_fold.txt

```diff
@@ -0,0 +1,38 @@
+select array_fold([],(x,y)->x+y-1);
+
+select array_fold([null],(x,y)->x+y-1);
```

Comment:
Test cases should be written in this format
```
query T
select array_transform([1, 2, NULL, 3], x -> x + 1)
----
[2,3,NULL,4]
```
You can see more examples of `array_transform` and `array_filter` here
https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/query/02_function/02_0061_function_array.test#L208
and please delete this text file and write them in the above test file, you can run `make sqllogic-test` in your local environment to see the result.